src/api/Awards/streak.js

```diff
@@ -0,0 +1,52 @@
+
+function checkStreak(match, playerName, streaks) {
+    let streak = streaks[playerName];
+      
+    if (!streak) {
+        streaks[playerName] = {
+            count: 1,
+            type: match.winner === playerName ? 'winning' : 'losing',
+            isOver: false
+        };
+    } else if (streak.isOver) {
+        // do nothing
+    } else if (streak.type === 'winning' && match.winner === playerName) {
+        streak.count += 1;
+    } else if (streak.type === 'losing' && match.loser === playerName) {
+        streak.count += 1;
+    } else {
+        streak.isOver = true;
+    }
+}
+
+// Assumes match results are sorted with most recent games first
+function calculate(matchResults) {
+    const awards = [];
+    let streaks = {};
+    
```

Comment:
Probably could do a quick sort here just so we aren't relying on the array being sorted correctly coming in. Using MomentJS something like:

```

function compareDatesDesc(a, b) {
    aDate = moment(a);
    bDate = moment(b);

    if (!aDate.isDate() || !bDate.isDate()) {
        throw new Error();
    }

    if (aDate.isAfter(bDate)) {
        return -1;
    } else if (aDate.isBefore(bDate)) {
        return 1;
    } else {
        return 0;
    }
}

matchResults.sort(compareDatesDesc);
```

I might have the logic backwards in the compare though, I haven't run the code.